client/src/components/Map/components/SuperBlock.js

```diff
@@ -79,21 +79,48 @@ export class SuperBlock extends Component {
                   .join('-') === blockDashedName
             )}
             key={blockDashedName}
+            noCurrentChallenge={noCurrentChallenge}
           />
         ))}
       </ul>
     );
   }
 
   render() {
-    const { superBlock, isExpanded, toggleSuperBlock } = this.props;
+    const {
+      superBlock,
+      isExpanded,
+      toggleSuperBlock,
+      nodes,
+      currentChallengeId
+    } = this.props;
+    const blocksForSuperBlock = nodes.filter(
+      node => node.superBlock === superBlock
+    );
+    const noCurrentChallenge = !nodes.some(
+      challenge => challenge.id === currentChallengeId
+    );
+    // If, somehow, the user does not have a current challenge, this opens the
+    // first superBlock.
+    const hasCurrentChallenge = noCurrentChallenge
```

Comment:
I don't understand this. You are using a `boolean` condition in a ternary to resolve a `boolean`?

Could you maybe describe what you are trying to achieve here?

Reply:
Do you mean what is `isOpen` for?  What I was trying to achieve was to get a block to be expanded if the current challenge is in it, but it also needs to respond to user clicks.

Originally I was worried that you would get weird behaviour in an edge case, but as I was writing out my explanation, I convinced myself it wasn't a problem.  So, I was wrong and booleans are fine after all!